vm/systemSmartContracts/esdt.go

```diff
@@ -1225,6 +1254,33 @@ func (e *esdt) setSpecialRole(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 	return vmcommon.Ok
 }
 
+func checkCorrectAddressForNFTCreateMultiShard(token *ESDTDataV2) error {
+	nftCreateRole := []byte(core.ESDTRoleNFTCreate)
+	mapLastBytes := make(map[uint8]struct{})
+	for _, esdtRole := range token.SpecialRoles {
+		hasCreateRole := false
```

Comment:
L1261-L1267 could have been another function.

Reply:
It is already a separated function :)